learning_journal_basic/setup.py

```diff
@@ -10,17 +10,30 @@
 
 requires = [
     'pyramid',
-    'pyramid_chameleon',
     'pyramid_debugtoolbar',
     'waitress',
+    'pyramid_jinja2',
+    'pyramid_tm',
+    'SQLAlchemy',
+    'transaction',
+    'zope.sqlalchemy',
+    'faker',
+    'ipython',
+    'pyramid-ipython'
     ]
 
 tests_require = [
     'WebTest >= 1.3.1',  # py3 compat
     'pytest',  # includes virtualenv
     'pytest-cov',
+    'pytest-watch',
+    'tox'
     ]
 
+development_extra = [
+    'ipython',
+    'pyramid_ipython'
+]
```

Comment:
You have development_extra as an optional requirements (good!) by no way to actually install these.  Take a look at how testing handles entry points.